src/dynamics/ads_player/ads_player.js

```diff
@@ -386,11 +386,28 @@ Scoped.define("module:Ads.Dynamics.Player", [
                     }
                     return true;
                 },
+                getMediaUrl: function(adObj) {
+                    let adMediaUrl = adObj?.data?.mediaUrl;
+                    if (adMediaUrl) {
+                        return adMediaUrl;
+                    }
+                    // backup plan if mediaurl is not in data object
+                    const mediaFiles = adObj?.data?.traffickingParameters && JSON.parse(adObj.data.traffickingParameters)?.mediaFiles;
+                    if (mediaFiles) {
```

Comment:
not mandatory if you think it will be always array, but for me will be better just in case check if mediaFiles are array.

Reply:
updated